validator/testdata/feature_tests/amp-image-slider.html

```diff
@@ -0,0 +1,42 @@
+<!--
+  Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS-IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the license.
+-->
+<!doctype html>
+<html ⚡>
+<head>
+  <meta charset="utf-8">
+  <link rel="canonical" href="./regular-html-version.html" />
+  <meta name="viewport" content="width=device-width,minimum-scale=1">
+  <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
+  <script async custom-element="amp-image-slider" src="https://cdn.ampproject.org/v0/amp-image-slider-0.1.js"></script>
+  <script async src="https://cdn.ampproject.org/v0.js"></script>
+  <style amp-custom>
+    .red {
+      color: red;
+    }
+    .green {
+      color: green;
+    }
+  </style>
+</head>
+<body>
+  <amp-image-slider tabindex="0" layout="responsive" width="200" height="100">
+    <amp-img src="/1.jpg" layout="fill"></amp-img>
+    <amp-img src="/2.jpg" layout="fill"></amp-img>
+    <div first class="red">BEFORE</div>
+    <div second class="green">AFTER</div>
+  </amp-image-slider>
```

Comment:
You should also test negative cases here and provide examples of where it shouldn't validate: 

e.g. a div with the first / second attribute not contained within `<amp-image-slider>`, as well as an `<amp-image-slider>` with less than 2 children.

Reply:
a div with the first / second attribute not contained
^ This could not be tested due to validator limitation. Will add others tho

Reply:
Since a `<div first>` has mandatory parent `<amp-image-slider>`, it should not be valid by itself (if it is not a child of `<amp-image-slider>`). We can add a test case to verify that.

Reply:
Addressed.